src/tvp/treeview.ts

```diff
@@ -0,0 +1,281 @@
+    private nvim: Neovim,
+    private config: WorkspaceConfiguration,
+    private buffer: Buffer,
+    private model: TreeModel,
+    private logger: log4js.Logger
+  ) { }
+
+  public async init(): Promise<void> {
+    await this.nvim.command(`silent file ${this.model.viewId}`)
+    const commandPrefix = "nnoremap <buffer> <silent>"
+    function cocCommand(param: string): string { return `:<C-u>CocCommand metals.tvp.view ${param} <CR>` }
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("toggleNode")} ${cocCommand("ToggleNode")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("forceChildrenReload")} ${cocCommand("ForceChildrenReload")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("gotoParentNode")} ${cocCommand("ParentNode")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("gotoFirstChild")} ${cocCommand("FirstSibling")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("gotoLastChild")} ${cocCommand("LastSibling")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("gotoPrevSibling")} ${cocCommand("PrevSibling")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("gotoLastSibling")} ${cocCommand("NextSibling")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("executeCommand")} ${cocCommand("ExecuteCommand")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("executeCommandAndOpenSplit")} ${cocCommand("ExecuteCommandAndOpenSplit")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("executeCommandAndOpenVSplit")} ${cocCommand("ExecuteCommandAndOpenVSplit")}`)
+    await this.nvim.command(`${commandPrefix} ${this.config.get<string>("executeCommandAndOpenTab")} ${cocCommand("ExecuteCommandAndOpenTab")}`)
+
+    await this.nvim.command(`au BufWinLeave <buffer> CocCommand metals.tvp.view Hidden ${this.model.viewId}`)
+
+    await this.nvim.command("setlocal nonumber norelativenumber nobuflisted nowrap noswapfile winfixwidth")// cursorline")
+    await this.nvim.command("setlocal statusline=%f buftype=nofile bufhidden=hide")
+
+    this.model.updateEvents(ev => this.handleModelUpdates(ev))
+
+    const viewsConfigs = this.config.get<TreeViewDescription[]>("initialViews")
+    await this.model.rootNode.expand()
+    const mbDesc = viewsConfigs.find(c => c.name === this.model.viewId)
+    if (mbDesc !== undefined) {
+      await Promise.all(mbDesc.expanded.map(parents => this.model.revealByParents(parents.concat(""))))
+      await this.nvim.call('coc#util#jumpTo', [0, 1])
+    }
+  }
+
+  public get bufferId(): number {
+    return this.buffer.id
+  }
+
+  private async method(window: Window, topLine: number, curLine: number): Promise<void> {
+    await window.setCursor([topLine, 1])
+    await window.setCursor([curLine, 1])
+  }
+
+  public handleModelUpdates(ev0: TreeModelUpdate): void {
```

Comment:
Not a big fan of `ev0`. Can we change this to something more explicit?

Reply:
Done.